applications/Diffy_chat.md

```diff
@@ -0,0 +1,133 @@
+
+## Development Roadmap :nut_and_bolt:
+
+We plan to execute 3 deliverables in two milestones:
+- a Substrate pallet for chat initiation;
+- a DOTRTC library for p2p transport implementation with test html pages for message passing;
+- a web-messenger dapp MVP with polkadot wallets authorisation.
+
+The project will be supported by a team of 2 developers, 1 UI/UX designer, 1 DevOps engineer and 1 QA.
+
+### Overview
+
+- **Total Estimated Duration:** 3 months
+- **Full-Time Equivalent (FTE):**  3,5 FTE
+- **Total Costs:** 30,000 USDT
+
+### Milestone 1 — Design and development of a pallet and a DOTRTC library
+
+- **Estimated duration:** 1,5 months
+- **FTE:**  2
+- **Costs:** 16,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how the developed features work. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure new functionality and robustness. In the guide, we will describe how to run these tests. |
+| 1. | Diffy chat pallet | We will develop a pallet (using Rust) which will include event handling to send and approve WebRTC offers for chat initiation. |
+| 2. | DOTRTC library | We will develop a DOTRTC library (with JS) for p2p transport implementation using a parachain with Diffy chat pallet. This library will provide an API for organizing p2p communication, will split packets into chunks (and reassemble on the receiver’s end) and provide methods for secure messaging between two participants (a sender will encrypt outgoing messages with recipient’s public key so only the recipient could decrypt them with his private key). The library will. |
+| 3. | HTML test page | We will deliver an HTML test page for DOTRTC library testing (setting a p2p channel between 2 users using the DOTRTC library). |
+
+### Milestone 2 — A web messenger MVP development
+
+- **Estimated duration:** 1,5 months
+- **FTE:**  1,5
+- **Costs:** 14,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Apache 2.0 |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how the new dapp works. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. | 
+| 0d. | Article | We will publish an **article** that explains what was achieved, how to use the new Dapp and what benefits what are the benefits of using the system |  
+| 1. | Contacts list feature | We will develop a contact list feature allowing users to tie real names to contact’s wallet addresses and store them encrypted in a blockchain. Encryption and decryption will be carried out on the frontend. | 
```

Comment:
What kind of encryption are you planning to use? Are you referring here to the frontend developed as part of milestone 1?